File: backend2/backend/ml-pipeline/service/pipeline_service.js
import { extractSkills } from '../pipelines/extract_skills.js';
import { rewriteResume } from '../pipelines/rewrite_resume.js';

export async function runPipeline({ job_description, resume }) {
  const extractedSkills = await extractSkills(job_description);
  const generatedResume = await rewriteResume(resume, extractedSkills);
  return { generated_resume: generatedResume };
}

File: ml_backend/backend/ml_pipeline/service/pipeline_service.py
from ..pipelines.extract_skills import extract_skills
from ..pipelines.rewrite_resume import rewrite_resume


def run_pipeline(job_description: str, resume: str) -> dict:
    extracted_skills = extract_skills(job_description)
    generated_resume = rewrite_resume(resume, extracted_skills)
    return {"generated_resume": generated_resume}
